test/unit/lodash.spec.js

```diff
@@ -0,0 +1,275 @@
+                } else {
+                    return randomNumber;
+                }
+            };
+            var testArray = _.fill(new Array(20), 1).map(testFunction);
+            var testArray2 = _.fill(new Array(10), 1).map(testFunction);
+            var params = datasets.map2(testArray);
+            var array = params[0];
+            var result = _.map.apply(_, params);
+            expect(array.length).to.eql(20);
+            expect(_.map.apply(_, datasets.map2(testArray))).to.eql(result);
+            params = datasets.map2(testArray2);
+            array = params[0];
+            expect(array.length).to.eql(10);
+            result = _.map.apply(_, params);
+            expect(_.map.apply(_, datasets.map2(testArray2))).to.eql(result);
+        });
+    });
+
+    describe('partition', function () {
+        it('should split persons to two tables using active property', function () {
+            var params = datasets.partition();
+            var array = params[0];
+            expect(array.length).to.eql(4);
+            expect(_.partition.apply(_, datasets.partition())).to.eql([[{name: 'Harry', age: 40, active: true},
+                {name: 'Francheska', age: 61, active: true}],
+                [{name: 'Andrinio', age: 36, active: false},
+                    {name: 'Kodi', age: 1, active: false}]
+            ]);
+        });
+        it('should split persons only to truthy table when predicate is true for all', function () {
+            var params = datasets.partition2();
+            var array = params[0];
+            expect(array.length).to.eql(4);
+            expect(_.partition.apply(_, datasets.partition2())).to.eql([[{name: 'Andrinio', age: 36, active: false},
+                {name: 'Harry', age: 40, active: true},
+                {name: 'Kodi', age: 1, active: false},
+                {name: 'Francheska', age: 61, active: true}],
+                []]);
+        });
+        it('should split persons only to falsy table when predicate is false for all', function () {
+            var params = datasets.partition3();
+            var array = params[0];
+            expect(array.length).to.eql(4);
+            expect(_.partition.apply(_, datasets.partition3())).to.eql([[],
+                    [{name: 'Andrinio', age: 36, active: false},
+                        {name: 'Harry', age: 40, active: true},
+                        {name: 'Kodi', age: 1, active: false},
+                        {name: 'Francheska', age: 61, active: true}]]
+            );
+        });
+    });
+
+    describe('reduce', function () {
+        it('should count sum of squared numbers', function () {
+            var params = datasets.reduce();
+            var array = params[0];
+            expect(array.length).to.eql(5);
+            expect(array).to.include(1);
+            expect(array).to.include(5);
+            expect(_.reduce.apply(_, datasets.reduce())).to.eql(55);
+        });
+    });
+
+    describe('sortBy', function () {
```

Comment:
sortBy: Once again you compare the same results in tests. The only thing you've tested is if array length is 2 for predicate, but what would happen if I were to put only nulls there? Next thing is that we need to test other predicate datatypes, too.